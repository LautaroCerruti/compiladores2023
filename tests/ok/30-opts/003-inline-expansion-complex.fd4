let b : Nat = (fix (aux: Nat->Nat->Nat) (x y :Nat) -> ifz y then print "x = " x else ((fun (f : Nat -> Nat -> Nat) -> (fun (a b : Nat) -> f x (y-1))) aux) 1 1) 32 5

let c : Nat = (fix (aux: Nat->Nat->Nat) (x y :Nat) -> ifz y then print "y = " y else (ifz 0 then aux x else (fun (z : Nat) -> z)) (y-1)) 2000 50

let d : Nat = (fix (aux: Nat->Nat->Nat) (x y :Nat) -> ((fun (f : Nat -> Nat -> Nat) -> (fun (a b : Nat) -> ifz y then print "a = " a else f x (y-1))) aux) 1 1) 32 1
