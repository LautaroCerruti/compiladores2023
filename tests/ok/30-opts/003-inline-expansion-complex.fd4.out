x = 32
y = 0
a = 1
